### ISSUE TITLE
Do Not Wait to Publish History During Simulation

### ISSUE THREAD

**Issue by jonjove** [stellar/stellar-core#2515]
Stellar-core does not publish the history checkpoint containing ledgers `[64*C, 64*C+64)` until ledger `64*C+64` is closed. As a specific example, the checkpoint which contains ledger 63 but not 64 is not published until after ledger 64 is closed.

This is annoying during simulation when you are trying to generate a history archive and then simulate from that history archive to the exact same ledger. As a specific example, suppose I want to generate a history archive that contains the checkpoint which contains ledger 63. Then I must close ledger 64 with that instance of stellar-core to trigger publish, but I will not be able to simulate to ledger 64 with another instance. It is impossible to get both instances to the same ledger, which can be desirable if you want to compare the databases.

We should add a flag that can change this behavior only for simulation.

**Comment by MonsieurNicolas** [comment]
Yeah we should not wait, but I don't think we should add a flag to do this: I would expect the behavior to be "never wait for simulation". User should not have to pick.
